### ISSUE TITLE
[feature] self-destructable\ephemeral posts

### ISSUE THREAD

**Issue by jk-na** [superseriousbusiness/gotosocial#349]
Pleroma offer an enhancement on the mastodon statuses endpoint which I found useful in the past, which enables the user to set an expiry date on a post.

In the pleroma version of the statuses endpoint the property in question is called `expires_at` and if a status posted to the API attributes an iso8601 datetime to the property then the associated status will automatically be deleted at that time, if the property is left blank, then it isn't.

I used this for my syndicated posts so that they would delete after 90 days, so not exactly ephemeral but an automated way of managing local content and DB size, it would be cool if I could do the same with gotosocial.

**Comment by tootbrute** [comment]
To build on this idea, I would even love to have this as a user level option, or server level option for all posts.

Maybe having all posts on the server get deleted after 6 months or a year? I know personally I think most posts lose their relevancy after time goes on.

I would be able to self-host gotosocial for more people if I didn't have to worry about disk space filling up with people's posts over the years. I guess this already exists as a script but I'm scared to mess around with self-hosting. I'm not that knowledgeable.

https://github.com/codl/forget

**Comment by tsmethurst** [comment]
Yeah this makes sense, it's something I wanna investigate when there's time.

**Comment by kensanata** [comment]
As a stop-gap measure, I've written a little Python script for myself:

https://src.alexschroeder.ch/gompotherium.git/

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/349.
